pkg/router/functionHandler.go

```diff
@@ -440,8 +478,70 @@ func (fh *functionHandler) tapService(fn *fv1.Function, serviceURL *url.URL) {
 	fh.executor.TapService(fn.ObjectMeta, fn.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType, *serviceURL)
 }
 
+func (fh functionHandler) parseStickinessCookie(request *http.Request) (*stickinessCookie, *http.Cookie, error) {
+	var stickinessCookie stickinessCookie
+	cookie, err := request.Cookie(STICKINESS_COOKIE)
+	if err != nil {
+		fh.logger.Debug("failed to get stickiness cookie value", zap.Error(err))
+		return nil, nil, err
+	}
+	cookieBytes, err := base64.StdEncoding.DecodeString(cookie.Value)
+	if err != nil {
+		fh.logger.Debug("failed to decode stickiness cookie value", zap.Error(err), zap.String("cookieValue", cookie.Value))
+		return nil, nil, err
+	}
+	err = json.Unmarshal(cookieBytes, &stickinessCookie)
+	if err != nil {
+		fh.logger.Debug("failed to unmarshal stickiness cookie", zap.Error(err), zap.String("cookieString", string(cookieBytes)))
+	}
+	return &stickinessCookie, cookie, err
+}
+
+func (fh functionHandler) invalidateStrictStickinessCookie(responseWriter http.ResponseWriter, cookie *http.Cookie) (int, error) {
+	fh.logger.Debug("invalid stickiness cookie (strict type)", zap.Any("cookie", cookie))
+	cookie.MaxAge = -1
+	responseWriter.Header().Add("Set-Cookie", cookie.String())
+	responseWriter.WriteHeader(http.StatusInternalServerError)
+	return responseWriter.Write([]byte("Invalid stickiness cookie (strict type)"))
+}
+
 func (fh functionHandler) handler(responseWriter http.ResponseWriter, request *http.Request) {
-	if fh.httpTrigger != nil && fh.httpTrigger.Spec.FunctionReference.Type == fv1.FunctionReferenceTypeFunctionWeights {
+
+	var stickinessCookie *stickinessCookie
+	if fh.stickinessParams.enabled {
+		maybeValid, cookie, err := fh.parseStickinessCookie(request)
+		if err == nil {
+			fn := fh.functionMap[maybeValid.FuncName]
+			isValid := false
+			if fn != nil {
+				isValid, err = fh.isValid(fn, maybeValid.SvcAddress)
+				if err != nil {
+					fh.logger.Error("request to executor isValid endpoint failed", zap.Error(err))
+					// TODO : write error to responseWrite and return response
+					return
+				}
+			}
+
+			if isValid {
+				fh.function = fn
+				stickinessCookie = maybeValid
+				fh.logger.Debug("chosen function backend's metadata (stickiness cookie)", zap.Any("metadata", fh.function))
+			} else if stickinessCookie.Type == STRICT {
+				// the cookie is strict and invalid so this request cannot be handled
+				fh.logger.Debug("expiring invalid stickiness cookie")
+				_, err = fh.invalidateStrictStickinessCookie(responseWriter, cookie)
+				if err != nil {
+					fh.logger.Error("error writing invalid stickiness cookie response", zap.Error(err))
+				}
+				return
+			}
+
+		} else {
+			fh.logger.Debug("failed to parse stickiness cookie, will be ignored", zap.Error(err))
```

Comment:
I think we should log some error, if not here, inside the parseStickinessCookie. Do you agree?

Reply:
This one will happen a lot as well, a couple of the debug calls in the function are good candiates for error level logging